packages/rfw/lib/src/flutter/argument_decoders.dart

```diff
@@ -779,6 +779,33 @@ class ArgumentDecoders {
   /// Extension mechanism for [gridDelegate].
   static final Map<String, ArgumentDecoder<SliverGridDelegate?>> gridDelegateDecoders = <String, ArgumentDecoder<SliverGridDelegate?>>{};
 
+  /// Returns a [SliverPersistentHeaderDelegate] from the specified map.
+  /// 
+  /// The `type` key specifies the kind of persistent header delegate.
+  /// 
+  /// Right now there is only one type `fixed` that creates a
+  /// [FixedSliverPersistentHeaderDelegate] using the keys `maxExtent`,
+  /// `minExtent` and `child`.
+  /// 
+  /// The types and defaults for these keys match the
+  /// identically named arguments to the default constructors of those classes.
+  /// 
+  /// If the type is not `fixed` or is missing, returns null.
+  static SliverPersistentHeaderDelegate? persistentHeaderDelegate(DataSource source, List<Object> key) {
+    final String? type = source.v<String>([...key, 'type']);
+    switch (type) {
+      case 'fixed':
+        return FixedSliverPersistentHeaderDelegate(
+          fixedMaxExtent: source.v<double>([...key, 'maxExtent']) ?? 32.0, 
+          fixedMinExtent: source.v<double>([...key, 'minExtent']) ?? 0.0, 
+          child: source.child(['child']),
+        );
+      case null:
+      default:
+        return null;
```

Comment:
can we do something like the other extension mechanisms in this file?

Reply:
Done. Thanks ;)